api/v1beta1/database_webhook.go

```diff
@@ -107,6 +130,52 @@ func ValidateSecretTemplates(templates map[string]string) error {
 	return nil
 }
 
+var helpers []string = []string{"Protocol", "Host", "Port", "Password", "Username", "Password", "Database"}
+var functions []string = []string{"Secret", "ConfigMap", "Query"}
+
+func ValidateTemplates(templates Templates) error {
+	for _, template := range templates {
+
+		// This regexp is getting fields from mustache templates so then they can be compared to allowed fields
+		reg := "{{\\s*\\.([\\w\\.]+)\\s*(.*?)\\s*}}"
+		r, _ := regexp.Compile(reg)
+
+		fields := r.FindAllStringSubmatch(template.Template, -1)
+		for _, field := range fields {
+			if validHelperField(field[1]) {
+				continue
+			} else if validFunctionField(field[1]) {
```

Comment:
think a nested if-if-else is not needed here. would maybe bring all to one level. But maybe I overlooked something?

```go
if validHelperField
  continue

if !validFunctionField
  err

if !ValidFunctionArg
  err

return nil
```

Current
```mermaid
  graph TD;
      for-loop
      for-loop --> validHelperField -- true --> next
      validHelperField -- false --> validFunctionField
      validFunctionField -- true --> !validFunctionArg -- true --> r:err
      !validFunctionArg -- false --> next
      validFunctionField -- false --> else
      else --> r:err
      next -- item --> for-loop
      next -- no item --> r:nil
```

same level
```mermaid
graph TD;
    for-loop
    for-loop --> validHelperField -- true --> next
    validHelperField --> !validFunctionField -- true --> r:err
    !validFunctionField --> !validFunctionArg -- true --> r:err2
    !validFunctionArg --> next
    next -- item --> for-loop
    next -- no item --> r:nil
```

Reply:
It seems so, I'll update. I think it's a leftover from the previous attempt

Reply:
I've checked, and I guess we can get rid of the first `if-else if`, but to support go template functions that should also be caught by the regexp, the arg checking should be happening only if the functionField was caught.

Reply:
Should be better now